### ISSUE TITLE
Doesn't check that binaries have been already downloaded

### ISSUE THREAD

**Issue by Enjeru** [nodkz/mongodb-memory-server#84]
Hi,

I got an issue that it is downloading binaries twice: after `npm install` and when instance is going to run.

I have a problem in this case:
In my CI/CD job I run inside docker container `npm i` and `npm test` commands and it always download binaries at least 1 time.
Please add checking that binaries have been already downloaded.

**Comment by nodkz** [comment]
It looks like you are using some exact mongodb version in your tests.

Please add `MONGOMS_VERSION` var to your env variables in CI config.

Related to https://github.com/nodkz/mongodb-memory-server/issues/76

**Comment by nodkz** [comment]
By default downloaded `latest` version

**Comment by Enjeru** [comment]
I am using the env variable in my `Dockerfile`. 
*Part of my Dockerfile:*
```
## Dockerfile
FROM node:8.9.3-slim

ENV MONGOMS_VERSION=3.4.4

COPY . /my-app/source

RUN rm -rf ./node_modules

RUN npm install --unsafe-perm --loglevel=warn
# Here we see downloading process

RUN npm test
#And here we see downloading process
```
I took a look on code and seems like in `MongoBinary` you create `cache` as empty object on startup without checking any existing downloads. And then the version is put to the cache object *only after downloading*.
But when docker container is running it doesn't have run `MongoInstance` and the cache object is still empty and it is a reason why it tries to download it again.

**Comment by AJRdev** [comment]
@nodkz Is the env variable MONGOMS_DOWNLOAD_DIR not fit for this specific use case ?

**Comment by nodkz** [comment]
Need to meet MONGOMS_DOWNLOAD_DIR and MONGOMS_VERSION. These vars are used for storing binaries in some path.

So by default with npm install will be downloaded `latest` version. But if tests uses for example v3.6.4 it will be fownloaded again.

What I think, that this lib should have some bin script, which can be used in docker file:
- firstly install npm
- then run cmd for install other mongo version if needed

**Comment by AJRdev** [comment]
@nodkz I've been trying to cache as well the mongdb binary download with Gitlab CI as well.
I've tried doing that by using the env variables you suggested (I'm using the lastest version of the mongodb binary so I didn't specified MONGOMS_VERSION), here is my gitlab ci file :

```bash
stages:
  - functionnal_tests

functionnal_tests:
 image: node:8
 stage: functionnal_tests
 variables:
  MONGOMS_DOWNLOAD_DIR: 'mongobinary'
 cache:
   paths:
   - mongobinary
   - node_modules
 script:
   - npm install
   - npm test
```
I've suceeded to cache the npm test command where the mongodb binary is downloaded to the path specified by MONGOMS_DOWNLOAD_DIR. 
But it seems that on the npm install command the binary is not downloaded to the same path, it is downloaded to **node_modules/mongodb-memory-server/ +   MONGOMS_DOWNLOAD_DIR**, here is the trace of the debug of mongdb-memory-server : 

```bash
$ npm install

> grpc@1.15.1 install /builds/CourtiersBordelais/api/node_modules/grpc
> node-pre-gyp install --fallback-to-build --library=static_library

node-pre-gyp WARN Using request for node-pre-gyp https download 
[grpc] Success: "/builds/CourtiersBordelais/api/node_modules/grpc/src/node/extension_binary/node-v57-linux-x64-glibc/grpc_node.node" is installed via remote

> bcrypt@3.0.1 install /builds/CourtiersBordelais/api/node_modules/bcrypt
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using request for node-pre-gyp https download 
[bcrypt] Success: "/builds/CourtiersBordelais/api/node_modules/bcrypt/lib/binding/bcrypt_lib.node" is installed via remote

> protobufjs@6.8.8 postinstall /builds/CourtiersBordelais/api/node_modules/protobufjs
> node scripts/postinstall


> mongodb-memory-server@2.4.3 postinstall /builds/CourtiersBordelais/api/node_modules/mongodb-memory-server
> node ./postinstall.js

mongodb-memory-server: checking MongoDB binaries cache...
Downloading MongoDB latest: 0 % (0mb / 84.5mb)
Downloading MongoDB latest: 52.2 % (44.1mb / 84.5mb)
mongodb-memory-server: binary path is /builds/api/node_modules/mongodb-memory-server/mongobinary/latest/mongod
```
Is this behaviour normal ?

**Comment by AJRdev** [comment]
I found out the source of the problem it was because of the path I gave to MONGOMS_DOWNLOAD_DIR that was incorrect and using the current directory, I specified the path of the build directory of the gitlab pipeline and I suceeded to cache the npm install and npm test binary download :)

But still why do we need to download two times and not only in the npm test command ?

**Comment by Enjeru** [comment]
Sorry for delay, I've investigated it deeper and when I set version as `latest` it works correct but if I provide another version it downloads twice. Seems like it needs to delete `version: 'latest'` from `postinstall.js` anyway it will use it by default.
About previous comment of install only in the `npm test` command in my case it nice to have this feature but if you will change it please leave a possibility to install it in `npm install` command may be just add env variable `MONGOMS_DOWNLOAD_ON_POSTINSTALL`

**Comment by Enjeru** [comment]
Can I add one more PR for `MONGOMS_DOWNLOAD_ON_POSTINSTALL` env variable?

**Comment by nodkz** [comment]
It should download binary automatically on post install, cause many users don't know how to increase timeout limit for test runners. So better to download binaries on package install.

Why do you need to add `MONGOMS_DOWNLOAD_ON_POSTINSTALL` variable?
You have any problems right now? Binary did not download as expected?

**Comment by Enjeru** [comment]
I found one more situation that I need to don't download in `npm i`. I would like to add env variable that will cancel this issue but leave it as it is by default. So in this case if you need to download in `npm i` you don't need to do anything, but with this variable you can avoid it.

**Comment by Enjeru** [comment]
Another words I have 2 cases when I need to download in `npm i` and another one when I need to download it in `npm test`

**Comment by nodkz** [comment]
So it can be named MONGOMS_DISABLE_POSTINSTALL

Open PR!
